core/src/main/java/org/infinispan/persistence/spi/CacheWriter.java

```diff
@@ -35,4 +39,30 @@
     * @throws PersistenceException in case of an error, e.g. communicating with the external storage
     */
    boolean delete(Object key);
+
+   /**
+    * Persist all provided entries to the store in a single batch update. If this is not supported by the
+    * underlying store, then entries are written to the store individually via {@link #write(MarshalledEntry)}.
+    * As the execution order of MarshalledEntries is not guaranteed, you should ensure that only a single entry exists
+    * in the Collection for a given key.
+    *
+    * @param entries a Collection of MarshalledEntry to be written to the store.
+    * @throws NullPointerException if entries is null.
+    */
+   default void writeBatch(Collection<MarshalledEntry<? extends K, ? extends V>> entries) {
```

Comment:
No, I was referring to the fact that Off Heap should make entries visible per address counter https://github.com/infinispan/infinispan/blob/master/core/src/main/java/org/infinispan/container/offheap/OffHeapDataContainer.java#L590

If these are batched internally by the Stream, I am not sure, but this would be an issue outside the scope of this.